openxr/src/foveation_profile_fb.rs

```diff
@@ -0,0 +1,76 @@
+use sys::BaseOutStructure;
+
+use crate::*;
+
+pub struct FoveationProfileFB {
+    inner: FoveationProfileFBInner,
+}
+
+pub struct FoveationLevelProfile {
+    pub level: FoveationLevelFB,
+    pub vertical_offset: f32,
+    pub dynamic: FoveationDynamicFB,
+}
+
+impl FoveationProfileFB {
+    pub unsafe fn create<G>(
+        session: Session<G>,
+        level_profile: Option<FoveationLevelProfile>,
+    ) -> Result<Self> {
+        let instance = session.instance();
+        let entry = instance.entry();
+        let fp = raw::FoveationFB::load(entry, instance.as_raw())?;
+
+        let mut level_profile = level_profile.map(|lp| {
+            let mut c = sys::FoveationLevelProfileCreateInfoFB::out(std::ptr::null_mut());
+            (*c.as_mut_ptr()).vertical_offset = lp.vertical_offset;
+            (*c.as_mut_ptr()).level = lp.level;
+            (*c.as_mut_ptr()).dynamic = lp.dynamic;
```

Comment:
Each of these is unsound, because they construct an `&mut T` addressing partially uninitialized data. The `out` method is intended for OpenXR *output* structures, i.e. structures that OpenXR populates; for structures we populate, use conventional Rust structure construction syntax.